### ISSUE TITLE
[Bug]: Assistance Required with Bottleneck in NatureCNN.forward() on GPU

### ISSUE THREAD

**Issue by jmSNU** [DLR-RM/stable-baselines3#2062]
### 🐛 Bug

I hope this message finds you well. I am encountering a significant bottleneck when calling the `NatureCNN.forward() `method in my code. The input tensor to this method has a shape of (B, 3, 64, 64), where B represents the batch size. Despite several attempts to debug, I have been unable to resolve the issue. Below are my observations:

1. The computation in `self.cnn(observations)` takes an indefinite amount of time to complete. I believe this issue is not related to my hardware, CUDA, cuDNN, or PyTorch setup, as the same code works as expected when executed in a standalone Python script under identical conditions.
2. I defined a simple dummy layer, `self.layer = nn.Conv2d(3, 32, kernel_size=8, stride=4, padding=0)`, within `NatureCNN.__init__().` However, this layer also hangs indefinitely when `self.layer(observations)` is executed in `NatureCNN.forward()`.
3. The problem disappears entirely if both the model and observations are moved to the CPU, suggesting that it is specifically related to GPU computation.

Given these findings, I would appreciate any insights or suggestions on how to address this issue. Thank you in advance for your time and assistance!

### To Reproduce

```
agent = SAC(policy_arch, env, verbose=1, **agent_cfg)
checkpoint_callback = CheckpointCallback(save_freq=1000, save_path=log_dir, name_prefix="model", verbose=2)
agent.learn(total_timesteps=n_timesteps, callback=checkpoint_callback) ## Stuck
```

### System Info

torch : 2.2.2
GPU : RTX 4090 24GB
CUDA : 12.6
Driver : 560.35.03
OS : Ubuntu 22.04
Python : 3.10
Stable Baselines3 : 2.3.2

### Checklist

- [X] My issue does not relate to a custom gym environment. (Use the custom gym env template instead)
- [X] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo
- [X] I have read the [documentation](https://stable-baselines3.readthedocs.io/en/master/)
- [X] I have provided a [minimal and working](https://github.com/DLR-RM/stable-baselines3/issues/982#issuecomment-1197044014) example to reproduce the bug
- [X] I've used the [markdown code blocks](https://help.github.com/en/articles/creating-and-highlighting-code-blocks) for both code and stack traces.

**Comment by jmSNU** [comment]
I hope this message helps anyone who may be facing a similar issue. Please make sure to check the cuDNN version, as it could be the cause of the problem."
